### PR TITLE
Fix warlock quests and trainers re #434

### PR DESCRIPTION
**Pull request by NickTyrer**
Fixes 1, 2, 3, 4, 5, 6, 8, 9, 10, 11, 12 and 14 in issue #434

### CODE REVIEW

**Comment by ratkosrb** [comment]
Use the batch script to create a new migration. Currently you have a database id different from the file name, which will make the server unable to start. It will complain that migration 20200409171600 is missing, because you are inserting 20200405163750 in the table instead.

For Windows:
https://github.com/vmangos/core/blob/development/sql/make_migration.bat
For Linux:
https://github.com/vmangos/core/blob/development/sql/touch_migration.sh

File: sql/migrations/20200409171600_world.sql [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+BEGIN
+DECLARE v INT DEFAULT 1;
+SET v = (SELECT COUNT(*) FROM `migrations` WHERE `id`='20200409171600');
+IF v=0 THEN
+INSERT INTO `migrations` VALUES ('20200409171600');
+-- Add your query below.
+
+
+--remove all incorrect trainers from An Imps Request (8419) Richard Kerwin (4565), Luther Pickman (4564), Ursula Deline (5495), Sandahl (5496), Thistleheart (5171), Alexander Calder (5173), Grol'dar (3324), Mirket (3325) and Kartosh (988)
+DELETE FROM mangos.creature_questrelation WHERE quest = 8419 AND id NOT IN(461,3326,4563);
```
Comment (Kittnz) [inline sql/migrations/20200409171600_world.sql]: don't need database ref here

Comment (NickTyrer) [reply sql/migrations/20200409171600_world.sql]: Cheers for the review, I have removed all database references.

**Comment by NickTyrer** [comment]
@ratkosrb this should be good to go, all issies fixed in #434 except number 13 (couldent figure how to fix that).

**Comment by ratkosrb** [comment]
That must be a condition on the gossip option.

**Pull request merged**